ui/app/components/dashboard/quick-actions-card.js

```diff
@@ -131,17 +146,19 @@ export default class DashboardQuickActionsCard extends Component {
 
   @action
   navigateToPage() {
-    let searchSelectParamRoute = this.searchSelectParams.route;
+    let route = this.searchSelectParams.route;
+    let param = this.paramValue.id;
 
     // kv has a special use case where if the paramValue ends in a '/' you should
     // link to different route
-    if (this.selectedEngine.type === 'kv') {
-      searchSelectParamRoute =
-        this.paramValue && this.paramValue?.endsWith('/')
-          ? 'vault.cluster.secrets.backend.list'
-          : 'vault.cluster.secrets.backend.show';
+    if (this.selectedEngine.type === KV) {
+      route =
+        this.paramValue?.path && this.paramValue?.path?.endsWith('/')
```

Comment:
there is a helper in utils that checks if it's a directory or not. To be consistent we should probably use that here too. It's called `import { pathIsDirectory } from 'kv/utils/kv-breadcrumbs';`

Reply:
once imported I believe this would be `pathIsDirectory(this.paramValue.path?)`, but you'll want to check in case path doesn't exists, not sure how that would work.

Reply:
Good call! I think that the helper checks to see if contains `/` so if it doesn't exist it will return false which will default to the details route